pilot/pkg/networking/core/v1alpha3/name_table.go

```diff
@@ -78,8 +78,15 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
 			if svc.Attributes.ServiceRegistry == string(serviceregistry.Kubernetes) &&
 				svc.Resolution == model.Passthrough && len(svc.Ports) > 0 {
 				for _, instance := range push.ServiceInstancesByPort(svc, svc.Ports[0].Port, nil) {
-					// Add individual addresses even for cross cluster.
-					if instance.Endpoint.SubDomain != "" && instance.Endpoint.Network == node.Metadata.Network {
+					if instance.Endpoint.Network != node.Metadata.Network {
```

Comment:
I think we have some semantic like "if network is unset it should access all networks" that we may need to preserve?

Reply:
That is a great point. I will add that